packages/components/src/components/Ansi/Ansi.js

```diff
@@ -0,0 +1,268 @@
+
+const Ansi = ({ children }) => {
+  let properties = {
+    foregroundColor: null,
+    backgroundColor: null,
+    bold: false,
+    italic: false,
+    underline: false,
+    conceal: false,
+    cross: false
+  };
+
+  let styles = {};
+  let text = '';
+  let line = [];
+
+  const reset = () => {
+    properties = {
+      foregroundColor: null,
+      backgroundColor: null,
+      bold: false,
+      italic: false,
+      underline: false,
+      conceal: false,
+      cross: false
+    };
+  };
+
+  const enable = flag => {
+    properties[flag] = true;
+  };
+
+  const disable = flag => {
+    properties[flag] = false;
+  };
+
+  const getXtermColor = commandStack => {
+    if (commandStack.length >= 2 && commandStack[0] === '5') {
+      commandStack.shift();
+      const colorIndex = +commandStack.shift();
+      if (colorIndex >= 0 && colorIndex <= 255) {
+        return colors[colorIndex];
+      }
+    }
+    return null;
+  };
+
+  const getTermColor = (colorIndex, type) => {
+    if (colorIndex > 7) {
+      return null;
+    }
+    if (type === 'l') {
+      return colors[colorIndex + 8];
+    }
+    return colors[colorIndex];
+  };
+
+  const setFGColor = (type, command) => {
+    properties.foregroundColor = getTermColor(type, command);
+  };
+
+  const setBGColor = (type, command) => {
+    properties.backgroundColor = getTermColor(type, command);
+  };
+
+  const setFGColor256 = commandStack => {
+    properties.foregroundColor = getXtermColor(commandStack);
+  };
+
+  const setBGColor256 = commandStack => {
+    properties.backgroundColor = getXtermColor(commandStack);
+  };
+
+  const setProperties = {
+    0: () => reset(),
+    1: () => enable('bold'),
+    3: () => enable('italic'),
+    4: () => enable('underline'),
+    8: () => enable('conceal'),
+    9: () => enable('cross'),
+
+    21: () => disable('bold'),
+    22: () => disable('bold'),
+    23: () => disable('italic'),
+    24: () => disable('underline'),
+    28: () => disable('conceal'),
+    29: () => disable('cross'),
+
+    30: () => setFGColor(0),
+    31: () => setFGColor(1),
+    32: () => setFGColor(2),
+    33: () => setFGColor(3),
+    34: () => setFGColor(4),
+    35: () => setFGColor(5),
+    36: () => setFGColor(6),
+    37: () => setFGColor(7),
+    38: s => setFGColor256(s),
+    39: () => setFGColor(9),
+
+    40: () => setBGColor(0),
+    41: () => setBGColor(1),
+    42: () => setBGColor(2),
+    43: () => setBGColor(3),
+    44: () => setBGColor(4),
+    45: () => setBGColor(5),
+    46: () => setBGColor(6),
+    47: () => setBGColor(7),
+    48: s => setBGColor256(s),
+    49: () => setBGColor(9),
+
+    90: () => setFGColor(0, 'l'),
+    91: () => setFGColor(1, 'l'),
+    92: () => setFGColor(2, 'l'),
+    93: () => setFGColor(3, 'l'),
+    94: () => setFGColor(4, 'l'),
+    95: () => setFGColor(5, 'l'),
+    96: () => setFGColor(6, 'l'),
+    97: () => setFGColor(7, 'l'),
+    99: () => setFGColor(9, 'l'),
```

Comment:
Is this needed? Similar for 109 below.

I can't find any reference to these codes being used. Looks like 39 and 49 would be used instead in these cases to reset the colours.

Reply:
You're right. I couldn't find the codes in the linux man pageshttps://man7.org/linux/man-pages/man4/console_codes.4.html. I will remove them